### PR TITLE
Remove `key` prop pass-thru on HeaderButton

### PR DESCRIPTION
**Pull request by lukebarnard1**
This is redundant and throws a warning

### CODE REVIEW

File: src/components/structures/RightPanel.js [hunk 1]
```diff
@@ -45,11 +45,7 @@ class HeaderButton extends React.Component {
         const TintableSvg = sdk.getComponent("elements.TintableSvg");
         const AccessibleButton = sdk.getComponent("elements.AccessibleButton");
 
-        return <AccessibleButton
-            className="mx_RightPanel_headerButton"
-            key={this.props.key}
-            onClick={ this.onClick }
-        >
+        return <AccessibleButton className="mx_RightPanel_headerButton" onClick={ this.onClick } >
```
Comment (t3chguy) [inline src/components/structures/RightPanel.js]: trailing space before `>` :(

Comment (t3chguy) [reply src/components/structures/RightPanel.js]: plus I feel like you at one point told me props get no space padding within `{}`

File: src/components/structures/RightPanel.js [hunk 2]
```diff
@@ -45,13 +45,9 @@ class HeaderButton extends React.Component {
         const TintableSvg = sdk.getComponent("elements.TintableSvg");
         const AccessibleButton = sdk.getComponent("elements.AccessibleButton");
 
-        return <AccessibleButton
-            className="mx_RightPanel_headerButton"
-            key={this.props.key}
-            onClick={ this.onClick }
-        >
+        return <AccessibleButton className="mx_RightPanel_headerButton" onClick={this.onClick} >
```
Comment (t3chguy) [inline src/components/structures/RightPanel.js]: /me thought should be `onClick={this.onClick}>` similar to the `<div` below not padding out its closing `>`

File: src/components/structures/RightPanel.js (hunk 1 above)
Comment (lukebarnard1) [reply src/components/structures/RightPanel.js]: this is true, I've added an eslint rule for this - https://github.com/matrix-org/matrix-react-sdk/pull/1436

Comment (t3chguy) [reply src/components/structures/RightPanel.js]: :D

**Pull request merged**